src/global_logic.js

```diff
@@ -933,6 +933,26 @@ module.exports.calcBasedOneSummon = function (summonind, prof, buff, totals) {
 
             expectedCycleDamagePerTurn = expectedCycleDamage / (expectedTurn + 1.0);
         }
+
+        var expectedLockoutTimePerTurn;
+        if (expectedTurn === Infinity) {
+            expectedLockoutTimePerTurn = 1.0 + (expectedAttack * 0.35);
+        } else {
+            // Base lockout + (Normal attack lockout * n times)
+            let expectedCycleLockoutTime = expectedTurn * (1.0 + (expectedAttack * 0.35));
+            // Ougi Lockout
+            expectedCycleLockoutTime += 1.0 + 0.35;
+            // Chainburst lockout
+            if (buff["chainNumber"] == 2) {
+                expectedCycleLockoutTime += 5.0 / buff["chainNumber"];
+            } else if (buff["chainNumber"] == 3) {
+                expectedCycleLockoutTime += 7.0 / buff["chainNumber"];
+            } else if (buff["chainNumber"] >= 4) {
+                expectedCycleLockoutTime += 9.0 / buff["chainNumber"] == 4 ? buff["chainNumber"] : 4;
+            }
```

Comment:
Can normalize the logic to formula? and Math.min for over chain
```js
let chainNumber = Math.min(4, buff["chainNumber"]);
if (chainNumber >= 2) {
    expectedCycleLockoutTime += (chainNumber * 2 + 1.0) / chainNumber;
}
```
- [x] Over chain ... ok
- [x] Skip chainNumber 1 (no chains)
- [x] Skip invalid case 0 or Negative chainNumver

Reply:
or should be function  getChainLockoutTime(chainNumber)
unit test can check chainNumer -1 .. 5 cases